benchmarks loops should be ignored

**Describe the bug**
Linter suggests to replace common benchmark loop.

```
internal/foo/engine/storage/memory/search_bench_test.go:54:2: for loop can be changed to use an integer range (Go 1.22+) (intrange)
	for i := 0; i < b.N; i++ {
```

Loop like `for i := 0; i < b.N; i++` is a standard go benchmark construct. 

Additionally, I think it's plain wrong to replace it with int range, because in that case `b.N` will be evaluated only once, while in for loop `b.N` is evaluated every iteration, and I think it is important as it allows the bench framework to adapt to execution time.

**To Reproduce**
Write a bench test.

**Expected behavior**
Linter should ignore such loops.

--- a/intrange.go
+++ b/intrange.go
@@ -86,6 +86,10 @@ func check(pass *analysis.Pass) func(node ast.Node) {
 
 		switch cond.Op {
 		case token.LSS: // ;i < x;
+			if isBenchmark(cond.Y) {
+				return
+			}
+
 			x, ok := cond.X.(*ast.Ident)
 			if !ok {
 				return
@@ -95,6 +99,10 @@ func check(pass *analysis.Pass) func(node ast.Node) {
 				return
 			}
 		case token.GTR: // ;x > i;
+			if isBenchmark(cond.X) {
+				return
+			}
+
 			y, ok := cond.Y.(*ast.Ident)
 			if !ok {
 				return
@@ -214,6 +222,28 @@ func check(pass *analysis.Pass) func(node ast.Node) {
 			Message: msg,
 		})
 	}
+}
+
+func isBenchmark(expr ast.Expr) bool {
+	selectorExpr, ok := expr.(*ast.SelectorExpr)
+	if !ok {
+		return false
+	}
+
+	if selectorExpr.Sel.Name != "N" {
+		return false
+	}
+
+	ident, ok := selectorExpr.X.(*ast.Ident)
+	if !ok {
+		return false
+	}
+
+	if ident.Name == "b" {
+		return true
+	}
+
+	return false
 }
 
 type bodyChecker struct {

--- a/testdata/main.go
+++ b/testdata/main.go
@@ -1,5 +1,7 @@
 package main
 
+import "testing"
+
 func main() {
 	for i := 2; i < 10; i++ {
 	}
@@ -124,5 +126,12 @@ func main() {
 	}
 
 	for i := 0; x > i; i = 0x1 + i { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
+	}
+
+	var b *testing.B
+	for i := 0; i < b.N; i++ {
+	}
+
+	for i := 0; b.N >= i; i++ {
 	}
 }

